src/bridgetree.rs

```diff
@@ -300,31 +247,57 @@ impl<H: Hashable + Clone> NonEmptyFrontier<H> {
 
 /// A possibly-empty Merkle frontier. Used when the
 /// full functionality of a Merkle bridge is not necessary.
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct Frontier<H, const DEPTH: u8> {
     frontier: Option<NonEmptyFrontier<H>>,
 }
 
+impl<H, const DEPTH: u8> TryFrom<NonEmptyFrontier<H>> for Frontier<H, DEPTH> {
+    type Error = FrontierError;
+    fn try_from(f: NonEmptyFrontier<H>) -> Result<Self, FrontierError> {
+        if f.position.max_altitude().0 <= DEPTH {
+            Ok(Frontier { frontier: Some(f) })
+        } else {
+            Err(FrontierError::MaxDepthExceeded {
+                altitude: f.position.max_altitude(),
+            })
+        }
+    }
+}
+
 impl<H, const DEPTH: u8> Frontier<H, DEPTH> {
     /// Constructs a new empty frontier.
     pub fn empty() -> Self {
         Frontier { frontier: None }
     }
 
-    /// Constructs a new frontier from a `[NonEmptyFrontier]`
+    /// Constructs a new non-empty frontier from is constituent parts.
     ///
-    /// Returns `None` if the provided frontier exceeds the maximum
-    /// allowed depth.
-    pub fn new(frontier: NonEmptyFrontier<H>) -> Option<Self> {
-        if frontier.size() > 1 << DEPTH {
-            None
-        } else {
-            Some(Frontier {
+    /// Returns `None` if the new frontier would exceed the maximum
+    /// allowed depth or if the list of ommers provided is not consistent
+    /// with the position of the leaf.
+    pub fn from_parts(
+        position: Position,
+        leaf: Leaf<H>,
+        ommers: Vec<H>,
+    ) -> Result<Self, FrontierError> {
+        if position.max_altitude().0 <= DEPTH {
+            NonEmptyFrontier::from_parts(position, leaf, ommers).map(|frontier| Frontier {
                 frontier: Some(frontier),
             })
+        } else {
+            Err(FrontierError::MaxDepthExceeded {
+                altitude: position.max_altitude(),
+            })
         }
```

Comment:
Slightly clearer and less fragile:
```suggestion
        NonEmptyFrontier::from_parts(position, leaf, ommers)
            .and_then(|frontier| frontier.try_into())
```